spec/src/modules/assistant.js

```diff
@@ -0,0 +1,265 @@
+/* eslint-disable no-unused-expressions, import/no-unresolved */
+import ConstructorIO from '../../../test/constructorio'; // eslint-disable-line import/extensions
+
+const dotenv = require('dotenv');
+const chai = require('chai');
+const sinon = require('sinon');
+const sinonChai = require('sinon-chai');
+const EventSource = require('eventsource');
+const { ReadableStream } = require('web-streams-polyfill');
+const qs = require('qs');
+const { createAssistantUrl, setupEventListeners } = require('../../../src/modules/assistant');
+const Assistant = require('../../../src/modules/assistant');
+const { encodeURIComponentRFC3986 } = require('../../../src/utils/helpers');
+const jsdom = require('../utils/jsdom-global');
+
+const bundled = process.env.BUNDLED === 'true';
+const bundledDescriptionSuffix = bundled ? ' - Bundled' : '';
+
+chai.use(sinonChai);
+dotenv.config();
+
+const testApiKey = process.env.TEST_REQUEST_API_KEY;
+const clientVersion = 'cio-mocha';
+
+const defaultOptions = {
+  apiKey: testApiKey,
+  version: clientVersion,
+  assistantServiceUrl: 'https://assistant.cnstrc.com',
+  clientId: '123',
+  sessionId: 123,
+};
+
+const defaultParameters = {
+  domain: 'assistant',
+};
+
+describe(`ConstructorIO - Assistant${bundledDescriptionSuffix}`, () => {
+  const jsdomOptions = { url: 'http://localhost' };
+  let cleanup;
+
+  beforeEach(() => {
+    cleanup = jsdom(jsdomOptions);
+    global.CLIENT_VERSION = clientVersion;
+    window.CLIENT_VERSION = clientVersion;
+
+    if (bundled) {
+      ConstructorIO.prop = window.ConstructorioClient;
```

Comment:
Thanks for looking into this and figuring out the fix for the failing tests. This is fixing it but I don't understand how. Do you mind adding a comment here explaining why we don't have to do this in other files but we have to do it here?